src/expression/mod.rs

```diff
@@ -30,6 +30,7 @@ pub struct UnaryExpressionData {
     pub e: Box<Expression>
 }
 
+// TODO Fix these enum variant names not to end with the enum name
```

Comment:
I was hesitant to outright fix the warning message generated for this one from the linter, given how much we use these names in the code. However, I think the linter's reasoning is valid. 

In short, it's too redundant and wordy to have an enum named `Expression`, and then to also have its variants include that same name in their names (i.e. `BinaryExpression`) when it's clear they're already some variety of the enum. I'm less sure of what to rename them to, unfortunately, but perhaps just `Binary` would be sufficient.

Here's a link to the linter warning: https://github.com/Manishearth/rust-clippy/wiki#enum_variant_names